iceoryx_systemtest/src/iox_publisher_systemtest.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/untyped_publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+
+#include <iostream>
+
+bool killswitch = false;
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // Register sigHandler for SIGINT
+    signal(SIGINT, sigHandler);
+    // we register a sighandler for SIGTERM to check if Roudi exits 
+    // the process correctly for the test
+    signal(SIGTERM, sigHandler);
+
+    std::cout << "Application iox_publisher_systemtest started" << std::endl;
+
+    iox::runtime::PoshRuntime::initRuntime("iox-ex-publisher-untyped");
+
+    iox::popo::UntypedPublisher publisher({"Radar", "FrontLeft", "Object"});
+    publisher.offer();
+
+    double ct = 0.0;
+    while (!killswitch)
```

Comment:
Why not write
```cpp
for(double ct = 0.0; !killswitch; ++ct)
```

then we would save two lines of code and it would still be understandable since everyone loves a `for` loop